tests/test_url.py

```diff
@@ -85,29 +85,36 @@ def test_sub(target: str, base: str, expected: str):
 
 
 def test_sub_with_different_schemes():
-    expected_error_msg = "Both URLs should have the same scheme"
+    expected_error_msg = r"^Both URLs should have the same scheme$"
     with pytest.raises(ValueError, match=expected_error_msg):
         URL("http://example.com/") - URL("https://example.com/")
 
 
 def test_sub_with_different_netlocs():
-    expected_error_msg = "Both URLs should have the same netloc"
+    expected_error_msg = r"^Both URLs should have the same netloc$"
     with pytest.raises(ValueError, match=expected_error_msg):
         URL("https://spam.com/") - URL("https://ham.com/")
 
 
 def test_sub_with_different_anchors():
-    expected_error_msg = "'path/to' and '/path' have different anchors"
+    expected_error_msg = r"^'path/to' and '/path' have different anchors$"
     with pytest.raises(ValueError, match=expected_error_msg):
         URL("path/to") - URL("/path/from")
 
 
 def test_sub_with_two_dots_in_base():
-    expected_error_msg = "'..' segment in '/path/..' cannot be walked"
+    expected_error_msg = r"^'..' segment in '/path/..' cannot be walked$"
     with pytest.raises(ValueError, match=expected_error_msg):
         URL("path/to") - URL("/path/../from")
 
 
+def test_sub_combined_with_join():
+    target = URL("http://example.com/path/to")
+    base = URL("http://example.com/")
+    rel = target - base
+    assert target == base.join(rel)
```

Comment:
```suggestion
    assert target == base / rel
```

Reply:
`URL.join(url)` and `/` are not interchangeable 😅;
`/` is the equivalent of `URL.joinpath(other, encoded=False)`;
So if you want to use `/`, you need to write `base / rel.path`.

Reply:
But even `base / rel.path` is not the same as `base.join(rel)`

Reply:
Ah, I see.. This was my attempt to suggest a proper round-trip data conversion so it could be made into a property-based test.